clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -1415,15 +1494,21 @@ void SYCLToolChain::AddImpliedTargetArgs(const llvm::Triple &Triple,
       ProcessElement(Elem);
   } else if (!HostTC.getTriple().isWindowsMSVCEnvironment()) {
     // If -ftarget-register-alloc-mode is not specified, the default is
-    // pvc:default on Windows and and pvc:auto otherwise.
-    StringRef DeviceName = "pvc";
-    StringRef BackendOptName = SYCL::gen::getGenGRFFlag("auto");
-    if (IsGen)
-      PerDeviceArgs.push_back({DeviceName, Args.MakeArgString(BackendOptName)});
-    else if (Triple.isSPIROrSPIRV() &&
-             Triple.getSubArch() == llvm::Triple::NoSubArch) {
-      BeArgs.push_back(Args.MakeArgString(RegAllocModeOptName + DeviceName +
-                                          ":" + BackendOptName));
+    // pvc:default on Windows and and pvc:auto otherwise when -device pvc is
+    // provided by the user.
+    ArgStringList TargArgs;
```

Comment:
nit: `ArgStringList TargArgs;` is declared 3 times at different places in this file. Is there a way to optimize them? Like adding them as a parameter in the function there are used?

Reply:
2 uses in the file are similar (gathering `-X*` options), the other is not related.  The similar entries can be consolidated but use is dependent on various behaviors so there is never a guarantee that the populated list would be used.  I can look into consolidating but would like to do it in a separate PR.